src/magic.cpp

```diff
@@ -300,6 +264,186 @@ void spell_type::load( const JsonObject &jo, const std::string & )
     }
 }
 
+void spell_type::serialize( JsonOut &json ) const
+{
+    json.start_object();
+
+    json.member( "type", "SPELL" );
+    json.member( "id", id );
+    json.member( "name", name.translated() );
```

Comment:
Maybe now's the time to add serialization function to `translation`? (And maybe add a global flag `json_editor_mode` and issue a debug message if the function is used outside the editor to avoid accidentally using the serialization for game saving.)

Reply:
i think what i'm going to do is wait until this PR lands (it'll probably be a bit) before adding such a function, then redo the spell_type members in favor of translation serialization.